internal/cmds/setting/argument/argument.go

```diff
@@ -0,0 +1,156 @@
+package argument
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/djdv/go-filesystem-utils/internal/cmds/setting/runtime"
+	"github.com/djdv/go-filesystem-utils/internal/generic"
+	"github.com/djdv/go-filesystem-utils/internal/parameter"
+)
+
+type (
+	// TODO: English
+	// Argument is the binding of a Parameter with a Go value.
+	// (The value is typically a pointer to a `Settings` struct member,
+	// but may be any settable value.)
+	Argument  generic.Couple[parameter.Parameter, any]
+	Arguments <-chan Argument
+
+	// TODO: rename - SetValue
+	// SetFunc should attempt to assign to each `Argument.ValueReference` it receives.
+	// (Typically by utilizing the `Argument.Parameter.Name` as a key to a value store.)
+	// SetFunc must send all unassigned `Argument`s (if any) to its output channel.
+	SetFunc func(context.Context, Arguments, ...TypeParser) (unsetArgs Arguments, _ <-chan error)
+
+	// TODO: rename ParseArgument
+	// ParseFunc receives a string representation of the data value,
+	// and returns a typed Go value of it.
+	ParseFunc func(argument string) (value any, _ error)
+
+	// TypeParser is the binding of a type with its corresponding parse function.
+	TypeParser struct {
+		reflect.Type
+		ParseFunc
+	}
+
+	errors = <-chan error
+)
+
+func argsFromSettings[
+	setPtr runtime.SettingsType[settings],
+	settings any,
+](ctx context.Context,
+	set setPtr,
+) (Arguments, errors, error) {
```

Comment:
Would it make sense to return a couple of [Arguments, errors] instead of two channels and an error?

Reply:
Addressed in da562368df7bfa7c3290adb800df43981a31b06e